src/Compilers/CSharp/Portable/Symbols/Attributes/SourceAttributeData.cs

```diff
@@ -184,7 +184,7 @@ internal SourceAttributeData WithOmittedCondition(bool isConditionallyOmitted)
             }
         }
 
-        [MemberNotNullWhen(true, nameof(AttributeClass), nameof(AttributeConstructor))]
```

Comment:
>nameof(AttributeClass) [](http://example.com/codeflow?start=33&length=22)

It looks like removal of ```AttributeClass``` is not necessary. We cannot have constructor without a type.

Reply:
The attribute adds no information for `AttributeClass`, as that API is already declared as non-nullable.